esp-hal/src/soc/psram_common.rs

```diff
@@ -23,3 +25,16 @@ impl PsramSize {
         matches!(self, PsramSize::AutoDetect)
     }
 }
+
+/// Returns the address range available in external memory.
+#[cfg(any(feature = "quad-psram", feature = "octal-psram"))]
+pub fn psram_range(_psram: &crate::peripherals::PSRAM) -> Range<usize> {
+    unsafe { super::MAPPED_PSRAM.memory_range.clone() }
+}
+
+/// Returns the address range available in external memory.
+#[cfg(any(feature = "quad-psram", feature = "octal-psram"))]
+pub fn psram_pointer(psram: &crate::peripherals::PSRAM) -> (*mut u8, usize) {
```

Comment:
Alternatively, we could collapse both these functions into one, by using `psram_range` that returns `Range<*mut u8>` containing the start and end.

It's a bit trickier to use as a end user though, it's a shame there isn't a `difference`/`size` method to give the difference between the start and end range easily.

Reply:
Sounds good, throwing out another name suggestion just for prosperity: `psram_raw_parts` just because it's used in a `core` infrequently, but I'm happy with either.

Reply:
`raw_parts` works, I decided against it because there is no "into_raw_parts", just "from_raw_parts". But we are also not the standard library, so I guess it's okay if the pattern doesn't apply to us :)